### PR TITLE
holdings: add request for holding in professional interface

### PR DESCRIPTION
**Pull request by vgranata**
- Allow librarian to request holding
- Adapted item-request component for both holding and item 

Co-Authored-by: Valeria Granata <valeria@chaw.com>

## Why are you opening this PR?

https://tree.taiga.io/project/rero21-reroils/task/2355

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: projects/admin/src/app/record/detail-view/document-detail-view/holding/default-holding-item/default-holding-item.component.ts [hunk 1]
```diff
@@ -94,9 +94,9 @@ export class DefaultHoldingItemComponent implements OnInit {
    * Add request on item and refresh permissions
    * @param itemPid - string
    */
-  addRequest(itemPid: string): void {
+  addRequest(recordPid: string,  recordType: string): void {
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/detail-view/document-detail-view/holding/default-holding-item/default-holding-item.component.ts:97]: double space after `,`

File: projects/admin/src/app/record/detail-view/document-detail-view/holding/default-holding-item/default-holding-item.component.ts [hunk 2]
```diff
@@ -94,9 +94,9 @@ export class DefaultHoldingItemComponent implements OnInit {
...
     const modalRef = this._modalService.show(ItemRequestComponent, {
-      initialState: { itemPid }
+      initialState: { recordPid , recordType }
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/detail-view/document-detail-view/holding/default-holding-item/default-holding-item.component.ts:99]: remove space before `,`

File: projects/admin/src/app/record/detail-view/document-detail-view/item-request/item-request.component.ts [hunk 3]
```diff
@@ -104,14 +115,24 @@ export class ItemRequestComponent implements OnInit {
...
+    if (this.recordType === 'item') {
+      key = 'item_pid';
+      body[key] = this.recordPid;
+    } else if (this.recordType === 'holding') {
+      key = 'holding_pid';
+      body[key] = this.recordPid;
+      key = 'description';
+      body[key] = model.description;
+    }
+    this._http.post('/api/' + this.recordType + '/request', body)
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/detail-view/document-detail-view/item-request/item-request.component.ts:135]: `/api/${this.recordType}/request`

**Review by Garfield-fr — APPROVED** [review]

File: projects/admin/src/app/record/detail-view/document-detail-view/item-request/item-request.component.ts (hunk 3 above)
Comment (zannkukai) [reply projects/admin/src/app/record/detail-view/document-detail-view/item-request/item-request.component.ts:135]: surrounding by "italic quote"

File: projects/admin/src/app/record/detail-view/holding-detail-view/serial-holding-detail-view/serial-holding-detail-view.component.ts [hunk 4]
```diff
@@ -110,7 +110,7 @@ export class SerialHoldingDetailViewComponent implements OnInit {
     this._recordService
       .getRecords(
         'items',
-        `holding.pid:${this.holding.id}`,
+        `holding.pid:${this.holding.id} AND NOT (type:provisional)`,
```
Comment (zannkukai) [inline projects/admin/src/app/record/detail-view/holding-detail-view/serial-holding-detail-view/serial-holding-detail-view.component.ts:113]: unnecessary parantheses

**Review by zannkukai — APPROVED** [review]

**Comment by Garfield-fr** [comment]
Title: `holdings: add request for holdings in professional interface`

**Pull request merged**